lib/streamlit/hello.py

```diff
@@ -14,57 +14,243 @@
 
-        new_rows = np.random.randn(10, 1)
+def data_frame_demo():
+    """
+    This demo shows how to use `st.write` to visualize Pandas DataFrames.
 
-        # Update status text.
-        status_text.text("The latest random number is: %s" % new_rows[-1, 0])
+    (Data courtesy of the [UN Data Exlorer](http://data.un.org/Explorer.aspx).)
+    """
+    import sys
+    import pandas as pd
+    if sys.version_info[0] < 3:
+        reload(sys)
+        sys.setdefaultencoding("utf-8")
 
-        # Append data to the chart.
-        chart.add_rows(new_rows)
+    @st.cache
+    def get_UN_data():
+        AWS_BUCKET_URL = "https://streamlit-demo-data.s3-us-west-2.amazonaws.com"
+        df = pd.read_csv(AWS_BUCKET_URL + "/agri.csv.gz")
+        return df.set_index("Region")
 
-        # Pretend we're doing some computation that takes time.
-        time.sleep(0.1)
+    try:
+        df = get_UN_data()
+    except urllib.error.URLError:
+        st.error("Connection Error. This demo requires internet access")
+        return
 
-    status_text.text("Done!")
-    st.balloons()
+    countries = st.multiselect("Choose countries", list(df.index),
+                               ["China", "United States of America"])
+    if not countries:
+        st.error("Please select at least one country.")
+        return
 
-    st.write(
-        """
-        PS: Want to know how we did this?
-        [You can learn about it here.](https://streamlit.io/docs/core_mechanics.html)
-    """
-    )
+    "### Gross Agricultural Production \[$1B\]"
+    st.write(df.loc[countries].sort_index())
+
+    ax = df.loc[countries].T.plot(kind="area", figsize=(20, 8), stacked=False)
+    ax.set_ylabel("GAP in Billions (Int $)", fontsize=25)
+    ax.set_xlabel("Year", fontsize=25)
+    ax.tick_params(labelsize=20)
+    ax.legend(loc=2, prop={"size": 20})
+    st.pyplot()
+
+
+DEMOS = OrderedDict({
+    "---": intro,
+    "Mapping Demo": mapping_demo,
+    "Animation Demo": fractal_demo,
+    "DataFrame Demo": data_frame_demo,
+    "Plotting Demo": plotting_demo,
+})
+
+def run():
+    demo_name = st.sidebar.selectbox("Choose a demo", list(DEMOS.keys()), 0)
+    demo = DEMOS[demo_name]
+
+    if demo_name != "---":
+        show_code = st.sidebar.checkbox("Show code", True)
+        st.markdown("# %s" % demo_name)
+        st.write(inspect.getdoc(demo))
+        demo()
+        if show_code:
+            st.markdown("## Code")
+            sourcelines, n_lines = inspect.getsourcelines(demo)
+            sourcelines = reset_indentation(remove_docstring(sourcelines))
+            st.code("".join(sourcelines))
+    else:
+        st.title("Welcome to Streamlit!")
+        st.write(
+            """
+            Streamlit is the best way to create bespoke apps.
+            """
+        )
+        demo()
+
+
+# This function parses the lines of a function and removes the docstring
+# if found. If no docstring is found, it returns None.
+def remove_docstring(lines):
```

Comment:
You could just not put docstrings in the functions above. I don't they're needed anyway.

Reply:
The way we do is we extract docstring and we show them on the screen.